python/tests/test_aac/plugins/root_schema_must_have_name/test_root_schema_must_have_name.py

```diff
@@ -9,22 +17,163 @@
         result = root_schema_has_name(definitions[0].instance, definitions[0], context.get_definitions_by_name("Schema")[0].instance)
         self.assertTrue(result.is_success())
         context.remove_definitions(definitions)
 
+
+    #  This test starts with a normal properly formed schema
+    #  It then takes the result from parse_and_load and specifically removes the 'name'
+    #  It then calls root_schema_has_name which will not find the name it is looking for
+    #  and return an ExecutionStatus.GENERAL_FAILURE from root_schema_has_name
     def test_root_schema_has_name_fail(self):
         context = LanguageContext()
-        definitions = context.parse_and_load(root_schema_without_name)
-        result = root_schema_has_name(definitions[0].instance, definitions[0], context.get_definitions_by_name("Schema")[0].instance)
-        self.assertFalse(result.is_success())
-        self.assertEqual(result.messages[0].source, "No file to reference")
-        context.remove_definitions(definitions)
+        try:
+            # Obtain the definitions for a properly formed schema
+            # This prevents any ParserErrors or LanguageErrors from being generated within parse_and_load
+            definitions = context.parse_and_load(root_schema_with_name)
+
+            # Now modify the fields to remove the 'name' field
+            # Most of this 'try' is copied from root_schema_has_name::_get_fields
+            try:
+                schema = definitions[0].instance
+                instance_fields = list[str]
+                context: LanguageContext = LanguageContext()
+                instance_fields: list[str] = []
+
+                for field in schema.fields:
+                    if field.name != 'name':
+                        # Copy the field over except
+                        # skip if it is 'name' to trigger the check in root_schema_has_name
+                        instance_fields.append(field)
+
+                if schema.extends:
+                    for ext in schema.extends:
+                        parent_schema = context.get_definitions_by_name(ext.name)
+                        if len(parent_schema) == 1:
+                            definitions[0].instance.fields.extend(_get_fields(parent_schema[0].instance))
+                ## Set fields to the modified list without a 'name' field
+                definitions[0].instance.fields = instance_fields
+            except:
+                traceback.print_exc()
+
+            result = root_schema_has_name(definitions[0].instance, definitions[0], context.get_definitions_by_name("Schema")[0].instance)
+            index = result.get_messages_as_string().find("must have a field named 'name'") # this is the message from line 61 of root_schema_has_name
+            if index != -1:
+                # Name message located
+                self.assertFalse(result.is_success())
+            else:
+                # Name message missing
+                self.assertFalse(True)
```

Comment:
I would think we would always want to actually be checking some kind of content/return with any of our assertions, not a manually set state?
```suggestion
                self.assertTrue(result.is_success())
```